### ISSUE TITLE
Compression Upgrade potential issue (also Supplementaries and Refined Storage are at fault)

### ISSUE THREAD

**Issue by ScoutTheSaneless** [P3pp3rF1y/SophisticatedStorage#166]
When I set up an auto-crafting set-up w Refined Storage for Supplementaries cages, the game just up and yeeted me out of my singleplayer world passing it off as a server issue and when I tried to get back it, it had an absolute screaming fit-

1. Set up Refined Storage w external cables connecting to netherite barrels (perhaps w a compacting upgrade)
2. Ask for a cage to be crafted or have one at ready
3. Await the cage to be crafted or throw it in
4. Watch as the game looses its mind
[Crash (Woah!) Log](https://github.com/P3pp3rF1y/SophisticatedStorage/files/11302228/crash-2023-04-21_23.00.47-server.txt)

**Comment by P3pp3rF1y** [comment]
I am unable to recreate refined storage not being able to auto craft from storage so I would need more info on that

And based on the crash it looks like something is causing the server start event to not happen before tile entities start to tick as it's supposed to. I can add a check for that not happening when it should be all that will do is allow the game to not crash and compacting won't work in that case.

If you can give me mc version on which you had this happen I can try to test with supplementaries but I suspect it's something else causing this so you will need to figure out the minimal set of mods with which it happens.

**Comment by ScoutTheSaneless** [comment]
1.18.2

**Comment by P3pp3rF1y** [comment]
I have retested in a pack with sophisticated storage, refined storage and supplementaries and I am unable to recreate neither the bug with crafting the cage nor the bug that happens for you when you join the world. So I will need more info on what other mod is causing this to happen / what steps I need to take to recreate the cage bug. 
As a side note I have fixed a bug with compression upgrade today with slot limits which potentially could have been the cage bug for you.
